cityofaustin/atd-service-bot
# Add Pipeline value from Zenhub to the DTS Portal's Knack App

# Intent

This PR hopes to close https://github.com/cityofaustin/atd-data-tech/issues/15878.

It adds support for keeping the `Pipeline` field in sync in the `DTS | Data and Technology Services Portal | PRODUCTION` knack app in sync with what ZenHub reports for each project index issue.

# Airflow

The airflow dag for this ETL now needs to pass in the ZenHub API key, and that change is [here](https://github.com/cityofaustin/atd-airflow/pull/216). 

# Tag-along changes

I have added a bare bones docker compose stack here to make spinning up a development environment for these easy and consistent. Not much too it, but you can get to a dev environment with just `docker compose run service-bot` now. 

# Testing

This is pretty easy to test locally.

* Pull down the branch
* Create a `.env` from the template, and populate these values in your environment file
* `docker compose run service-bot` should drop you into a bash terminal inside the dev container.
* `./gh_index_issues_to_dts_portal.py`
  * This will update production if there are any changes, and if there are none, then it will report that there's nothing to do. If you want, you can change the pipeline status of a project index to sync it, but please change it back and sync again if you do.
